core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -441,7 +446,34 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
         m_league_size(arg_policy.league_size()),
         m_team_size(arg_policy.team_size()),
         m_vector_size(arg_policy.impl_vector_length()) {
-    // FIXME_SYCL check_parameters
+    m_shmem_size =
+        (m_policy.scratch_size(0, m_team_size) +
+         m_policy.scratch_size(1, m_team_size) +
+         FunctorTeamShmemSize<FunctorType>::value(m_functor, m_team_size));
```

Comment:
Wait are you trying to get both lvl 1 and lvl 0 scratch from the actual sycl shmem which is likely limited by GPU shmem? That is not the intend. I'd rather have you say lvl 1 max size is zero instead until that is implemented.

Reply:
See, #3763 for the pull request corresponding to the new TeamPolicy implementation. Sorry for the confusion!